### ISSUE TITLE
no-useless-escape incompatible with unicode-aware regex litterals

### ISSUE THREAD

**Issue by guimier** [eslint/eslint#7877]
* **ESLint Version:** 3.12.2
* **Node Version:** 7.3.0
* **npm Version:** 3.10.10
* **parser:** default

`.eslintrc.json`
```json
{
    "env": {
        "es6": true
    },
    "rules": {
        "no-useless-escape": "error"
    }
}
```

`lint.js`:
```js
const re = /\]/u
```

`syntax.js`:
```js
const re = /]/u
```

**What did you expect to happen?**
Given `syntax.js` is not valid (see output below, not a parser bug afaict), `lint.js` should be accepted by the rule.

**What actually happened?** (exact output except path removal)
`eslint lint.js syntax.js`
```
/…/lint.js
  1:13  error  Unnecessary escape character: \]  no-useless-escape

/…/syntax.js
  1:13  error  Parsing error: Error parsing regular expression: Invalid regular expression: /]/: Lone quantifier brackets

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. I think this is a duplicate of https://github.com/eslint/eslint/issues/7789, which has already been fixed on master but hasn't made it into a release yet.

We have a release scheduled for today, so the fix should be available quite soon.

**Comment by guimier** [comment]
Great, thank you.
